src/OpenLoco/src/Vehicles/Vehicle.h

```diff
@@ -51,6 +51,20 @@ namespace OpenLoco::Vehicles
         constexpr uint8_t isStillPowered = 1 << 1;
     }
 
+    enum class WaterMotionFlags : uint32_t
+    {
+        none = 0U,
+        isStopping = 1U << 0,
+        isLeavingDock = 1U << 1,
+        hasReachedDock = 1U << 16,
+        hasReachedADestination = 1U << 17,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(WaterMotionFlags);
+    constexpr bool hasFlags(WaterMotionFlags flags, WaterMotionFlags flagsToTest)
+    {
+        return (flags & flagsToTest) != WaterMotionFlags::none;
+    }
```

Comment:
This is best moved into VehicleHead.cpp and also renamed as hasFlags is a bit too generic for a free standing function.